sql/catalyst/src/main/scala/org/apache/spark/sql/util/SQLOpenHashSet.scala

```diff
@@ -60,6 +61,57 @@ class SQLOpenHashSet[@specialized(Long, Int, Double, Float) T: ClassTag](
 }
 
 object SQLOpenHashSet {
+  def withNullCheckFunc(
+      dataType: DataType,
+      hashSet: SQLOpenHashSet[Any],
+      handleNotNull: Any => Unit,
+      handleNull: () => Unit): ArrayData => Int => Unit = {
+    (array: ArrayData) => {
+      (index: Int) =>
+        if (array.isNullAt(index)) {
+          if (!hashSet.containsNull) {
+            hashSet.addNull
+            handleNull()
+          }
+        } else {
+          val elem = array.get(index, dataType)
+          handleNotNull(elem)
+        }
+    }
+  }
+
+  def withNullCheckCode(
+      arrayType1: DataType,
+      arrayType2: DataType,
+      hashSet: String,
+      handleNotNull: String => String => String,
+      handleNull: String): String => String => String = {
```

Comment:
ditto, I don't think the currying syntax is needed here.

Reply:
Changed